src/displayBackend/wayland/Display.cpp

```diff
@@ -433,13 +448,20 @@ void Display::registryHandler(wl_registry *registry, uint32_t id,
 	}
 #endif
 #ifdef WITH_ZCOPY
-	if (interface == "wl_drm")
-	{
-		mWaylandDrm.reset(new WaylandDrm(registry, id, version));
-	}
-	if (interface == "wl_kms")
+	if (!mDisableZCopy)
 	{
-		mWaylandKms.reset(new WaylandKms(registry, id, version));
+		if (interface == "wl_drm")
+		{
+			mWaylandDrm.reset(new WaylandDrm(registry, id, version));
+		}
+		if (interface == "wl_kms")
+		{
+			mWaylandKms.reset(new WaylandKms(registry, id, version));
+		}
+		if (interface == "zwp_linux_dmabuf_v1")
+		{
+			mWaylandLinuxDmabuf.reset(new WaylandLinuxDmabuf(registry, id, version));
```

Comment:
I may ask wrong questing, but why *reset* method is called here, wouldn't the following work?

`mWaylandLinuxDmabuf = WaylandLinuxDmabufPtr(new WaylandLinuxDmabuf(registry, id, version));`

Reply:
See [reset](http://www.cplusplus.com/reference/memory/shared_ptr/reset/) description